src/modding/ModManager.cs

```diff
@@ -857,6 +1308,251 @@ private void OpenInfoUrlPressed()
         }
     }
 
+    private void OnDependencyPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        if (selectedMod is null)
+        {
+            return;
+        }
+
+        otherModInfoDialog.WindowTitle = TranslationServer.Translate("MOD_DEPENDENCIES");
+        var infoText = string.Empty;
+
+        var currentModDependencies = selectedMod.Info.Dependencies;
+        if (currentModDependencies != null)
+        {
+            foreach (string currentDependency in currentModDependencies)
+            {
+                if (!string.IsNullOrWhiteSpace(currentDependency))
+                {
+                    infoText += "* " + currentDependency + "\n";
+                }
+            }
+        }
+        else
+        {
+            infoText += TranslationServer.Translate("NO_MOD_DEPENDENCIES");
+        }
+
+        otherModInfoDialog.DialogText = infoText;
+        otherModInfoDialog.PopupCenteredShrink();
+    }
+
+    private void OnRequiredModsPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        if (selectedMod is null)
+        {
+            return;
+        }
+
+        otherModInfoDialog.WindowTitle = TranslationServer.Translate("MOD_REQUIRED_MODS");
+        var infoText = string.Empty;
+
+        var currentModRequiredMods = selectedMod.Info.RequiredMods;
+        if (currentModRequiredMods != null)
+        {
+            foreach (string currentRequiredMod in currentModRequiredMods)
+            {
+                if (!string.IsNullOrWhiteSpace(currentRequiredMod))
+                {
+                    infoText += "* " + currentRequiredMod + "\n";
+                }
+            }
+        }
+        else
+        {
+            infoText += TranslationServer.Translate("NO_REQUIRED_MODS");
+        }
+
+        otherModInfoDialog.DialogText = infoText;
+        otherModInfoDialog.PopupCenteredShrink();
+    }
+
+    private void OnIncompatiblePressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        if (selectedMod is null)
+        {
+            return;
+        }
+
+        otherModInfoDialog.WindowTitle = TranslationServer.Translate("INCOMPATIBLE_WITH");
+        var infoText = string.Empty;
+
+        var currentModIncompatibleMods = selectedMod.Info.IncompatibleMods;
+        if (currentModIncompatibleMods != null)
+        {
+            foreach (string currentIncompatibleMod in currentModIncompatibleMods)
+            {
+                if (!string.IsNullOrWhiteSpace(currentIncompatibleMod))
+                {
+                    infoText += "* " + currentIncompatibleMod + "\n";
+                }
+            }
+        }
+        else
+        {
+            infoText += TranslationServer.Translate("NO_MOD_INCOMPATIBLE");
+        }
+
+        otherModInfoDialog.DialogText = infoText;
+        otherModInfoDialog.PopupCenteredShrink();
```

Comment:
This code seems duplicated quite many times. I'd make this a helper method that takes in as parameters the things that are different for the different boxes.